lib/net/ssh/service/forward.rb

```diff
@@ -105,10 +108,12 @@ def local(*args)
     #   ssh.forward.cancel_local(1234)
     #   ssh.forward.cancel_local(1234, "0.0.0.0")
     def cancel_local(port, bind_address="127.0.0.1")
+      channel_id = session.lookup_channel_id_from_localport(port)
       socket = @local_forwarded_ports.delete([port, bind_address])
       socket.shutdown rescue nil
       socket.close rescue nil
       session.stop_listening_to(socket)
+      session.channels.delete(channel_id)
```

Comment:
i don't think this is neccesary such channels will be eventually closed and close will remove them from session. Also there can be multiple channels for a given port, so closing just the first doesn't sounds right